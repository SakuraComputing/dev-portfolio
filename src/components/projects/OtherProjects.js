import React from 'react';
import OtherProjectItem from '../../components/projects/OtherProjectItem';

class OtherProjects extends React.Component {

    render() {

        const { otherprojects } = this.props;

        let otherProjectItems;

        if(otherprojects.length > 0) {
            otherProjectItems = otherprojects.map(otherproject => (
                <OtherProjectItem key={otherproject.id} otherproject={otherproject} />
            ));
        }

        return(
            <section className="projects section">
                <div className="section-inner">
                    <h2 className="heading">Other Projects</h2>
                    <div className="content">
                        {otherProjectItems}
                        <p></p>
                        <a className="btn btn-cta-secondary" href="#">More on CoderWall <i className="fa fa-chevron-right"></i></a>
                    </div>
                </div>
            </section>
        )
    }
}
export default OtherProjects;

