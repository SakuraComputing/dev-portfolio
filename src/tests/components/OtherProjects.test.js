import React from 'react';
import { shallow } from 'enzyme';
import OtherProjects from '../../components/OtherProjects';

describe('Other Projects Section', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<OtherProjects />);
        expect(wrapper).toMatchSnapshot();
    });
});