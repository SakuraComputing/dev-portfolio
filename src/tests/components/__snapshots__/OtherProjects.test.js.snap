// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Other Projects Section should render correctly 1`] = `
<section
  className="projects section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Other Projects
    </h2>
    <div
      className="content"
    >
      <div
        className="item"
      >
        <h3
          className="title"
        >
          <a
            href="#"
          >
            Project Title Cras Lacus
          </a>
        </h3>
        <p
          className="summary"
        >
          Provide a brief description of your project.
        </p>
        <p>
          <a
            className="more-link"
            href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-mobile-apps-delta/"
            target="_blank"
          >
            <i
              className="fa fa-external-link"
            />
             Find out more
          </a>
        </p>
      </div>
      <div
        className="item"
      >
        <h3
          className="title"
        >
          <a
            href="#"
          >
            Project Title Cras eget lacus
          </a>
           
          <span
            className="label label-theme"
          >
            Open Source
          </span>
        </h3>
        <p
          className="summary"
        >
          Provide a brief description of your project.
        </p>
        <p>
          <a
            className="more-link"
            href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-mobile-apps-delta/"
            target="_blank"
          >
            <i
              className="fa fa-external-link"
            />
             View on GitHub
          </a>
        </p>
      </div>
      <div
        className="item"
      >
        <h3
          className="title"
        >
          <a
            href="#"
          >
            Project Title Maecenas Ornare
          </a>
           
          <span
            className="label label-theme"
          >
            Open Source
          </span>
        </h3>
        <p
          className="summary"
        >
          Provide a brief description of your project.
        </p>
        <p>
          <a
            className="more-link"
            href="http://themes.3rdwavemedia.com/website-templates/responsive-bootstrap-theme-for-mobile-apps-delta/"
            target="_blank"
          >
            <i
              className="fa fa-external-link"
            />
             View on GitHub
          </a>
        </p>
      </div>
      <a
        className="btn btn-cta-secondary"
        href="#"
      >
        More on CoderWall 
        <i
          className="fa fa-chevron-right"
        />
      </a>
    </div>
  </div>
</section>
`;
